src/tests/resistance/fuzzing.rs

```diff
@@ -398,6 +820,143 @@ async fn fuzzing_slightly_corrupted_version_during_handshake_responder_side() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn fuzzing_slightly_corrupted_version_when_node_initiates_handshake() {
+    // ZG-RESISTANCE-003 (part 4, version only)
+    //
+    // This particular case is considered alone because it is at particular risk of causing
+    // troublesome behaviour, as seen with the valid metadata fuzzing against zebra.
+    //
+    // zebra: disconnects immediately.
+    // zcashd: Some messages get ignored and timeout. Most cause an immedietely due to
+    //          - main: PROCESSMESSAGE: INVALID MESSAGESTART, or
+    //          - net: Oversized message from peer
+
+    let locked_rng = Arc::new(Mutex::new(seeded_rng()));
```

Comment:
I'm not sure there's a different way of doing this?

The issue is that I can't generate the payloads ahead of time (since the payload relies on both the node and peer addresses). This means `rng` is needed inside each future / concurrently.

Overall not an issue, just something I'm happy to know if I did correctly.

Reply:
I don't see many solutions, this might be the best with the current setup.